src/Control/Carrier/FossaApiClientIO.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE GADTs #-}
+
+module Control.Carrier.FossaApiClientIO (FossaApiClientIOC, runFossaApiClientIO) where
+
+import App.Fossa.FossaAPIV1 qualified as API
+import Control.Algebra (Has)
+import Control.Carrier.Reader (ReaderC, runReader)
+import Control.Carrier.Simple (SimpleC, interpret)
+import Control.Effect.Diagnostics (Diagnostics)
+import Control.Effect.FossaApiClient (FossaApiClientF (..))
+import Control.Effect.Lift (Lift)
+import Control.Effect.Reader (Reader, ask)
+import Effect.Logger (Logger, logDebug)
+import Fossa.API.Types (ApiOpts, Organization)
+
+-- | A carrier to run Fossa API functions in the IO monad
+type FossaApiClientIOC m = SimpleC FossaApiClientF (ReaderC ApiOpts m)
+
+-- | Runs FossaAPI effects as IO operations
+--
+-- TODO: Flexible exception handling.  Currently exceptions are raised as
+-- Diagnostic errors.
+runFossaApiClientIO ::
+  ( Has (Lift IO) sig m
+  , Has Diagnostics sig m
+  , Has Logger sig m
+  ) =>
+  ApiOpts ->
+  FossaApiClientIOC m a ->
+  m a
+runFossaApiClientIO apiOpts =
+  runReader apiOpts
+    . interpret
+      ( \case
+          GetOrganization -> getOrganization
+          GetApiOpts -> pure apiOpts
+      )
```

Comment:
I actually can't quite do that here because it's a point-free function and there's another argument expected.

Reply:
probably you can just use `.` then

Reply:
Why is that not a problem for the other simple effects like ReadFS?

Reply:
Actually, I think there's an issue with using `SimpleC FooF (BarC Baz m)`

Notably, this works perfectly:
```haskell
type FossaApiClientIOC = SimpleC FossaApiClientF

runFossaApiClientIO ::
  ( Has (Lift IO) sig m
  , Has Diagnostics sig m
  , Has Logger sig m
  ) =>
  ApiOpts ->
  FossaApiClientIOC (ReaderC ApiOpts m) a ->
  m a
runFossaApiClientIO apiOpts = runReader apiOpts . runFossaApiClientIO'

runFossaApiClientIO' ::
  ( Has (Lift IO) sig m
  , Has Diagnostics sig m
  , Has (Reader ApiOpts) sig m
  , Has Logger sig m
  ) =>
  FossaApiClientIOC m a ->
  m a
runFossaApiClientIO' = interpret $ \case
  GetOrganization -> getOrganization
  GetApiOpts -> ask @ApiOpts
```

Reply:
That's because by putting it in it's own function you are basically putting parentheses around it.  There isn't a pretty way using just `$` is all.  `interpret` is a function of two arguments, so `runReader apiOpts . interpret` isn't going to work.